feature/bgp/otg_tests/bgp_override_as_path_split_horizon_test/bgp_override_as_path_split_horizon_test.go

```diff
@@ -0,0 +1,581 @@
+		SetAddress(advertisedRoutesv4Net).
+		SetPrefix(uint32(advertisedRoutesv4Prefix)).
+		SetCount(1)
+
+	bgpNeti1AsPath := bgpNeti1Bgp4PeerRoutes.AsPath().SetAsSetMode(gosnappi.BgpAsPathAsSetMode.INCLUDE_AS_SET)
+	bgpNeti1AsPath.Segments().Add().SetAsNumbers(asSeg).SetType(gosnappi.BgpAsPathSegmentType.AS_SEQ)
+
+	t.Logf("Pushing config to OTG and starting protocols...")
+	args.otg.PushConfig(t, args.otgConfig)
+	time.Sleep(30 * time.Second)
+	args.otg.StartProtocols(t)
+	time.Sleep(30 * time.Second)
+}
+
+// verifyPrefixesTelemetry confirms that the dut shows the correct numbers of installed,
+// sent and received IPv4 prefixes.
+func verifyPrefixesTelemetry(t *testing.T, dut *ondatra.DUTDevice, nbr string, wantInstalled, wantSent uint32) {
+	t.Helper()
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	prefixesv4 := statePath.Neighbor(nbr).AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Prefixes()
+	if gotInstalled := gnmi.Get(t, dut, prefixesv4.Installed().State()); gotInstalled != wantInstalled {
+		t.Errorf("Installed prefixes mismatch: got %v, want %v", gotInstalled, wantInstalled)
+	}
+	if gotSent := gnmi.Get(t, dut, prefixesv4.Sent().State()); gotSent != wantSent {
+		t.Errorf("Sent prefixes mismatch: got %v, want %v", gotSent, wantSent)
+	}
+}
+
+// configreRoutePolicy adds route-policy config.
+func configureRoutePolicy(t *testing.T, dut *ondatra.DUTDevice, name string, pr oc.E_RoutingPolicy_PolicyResultType) {
+	d := &oc.Root{}
+	rp := d.GetOrCreateRoutingPolicy()
+	pd := rp.GetOrCreatePolicyDefinition(name)
+	st, err := pd.AppendNewStatement("id-1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	stc := st.GetOrCreateConditions()
+	stc.InstallProtocolEq = oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP
+	st.GetOrCreateActions().PolicyResult = pr
+	gnmi.Replace(t, dut, gnmi.OC().RoutingPolicy().Config(), rp)
+}
+
+// verifyOTGPrefixTelemetry is to Validate prefix received on OTG por2.
+func verifyOTGPrefixTelemetry(t *testing.T, otg *otg.OTG, wantPrefix bool) {
+	t.Helper()
+	_, ok := gnmi.WatchAll(t, otg, gnmi.OTG().BgpPeer(atePort2.Name+".BGP4.peer").UnicastIpv4PrefixAny().State(),
+		time.Minute, func(v *ygnmi.Value[*otgtelemetry.BgpPeer_UnicastIpv4Prefix]) bool {
+			return v.IsPresent()
+		}).Await(t)
+
+	if ok {
+		bgpPrefixes := gnmi.GetAll(t, otg, gnmi.OTG().BgpPeer(atePort2.Name+".BGP4.peer").UnicastIpv4PrefixAny().State())
+		for _, prefix := range bgpPrefixes {
+			if prefix.GetAddress() == advertisedRoutesv4Net {
+				if wantPrefix {
+					gotASPath := prefix.AsPath[len(prefix.AsPath)-1].GetAsNumbers()
+					t.Logf("Received prefix %v on otg as expected with AS-PATH %v", prefix.GetAddress(), gotASPath)
+				} else {
+					t.Errorf("Prefix %v is not received on otg", prefix.GetAddress())
+				}
+			}
+		}
+	}
+}
+
+func testSplitHorizonNoAllowOwnIn(t *testing.T, args *otgTestArgs) {
+	t.Log("Baseline Test No allow-own-in")
```

Comment:
Here and throughout, update the log message for the test to match the subtest heading in the readme. (ie: 1.54.1  Test no allow-own-in)

Reply:
This is updated.